packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/EditView/index.js

```diff
@@ -175,9 +177,12 @@ const ApiTokenCreateView = () => {
             permissions: body.type === 'custom' ? state.selectedActions : null,
           });
 
-      if (isCreating) {
+      if (isCreating && !canOnlyCreate) {
         history.replace(`/settings/api-tokens/${response.id}`, { apiToken: response });
         setCurrentStep('apiTokens.success');
+      } else if (canOnlyCreate) {
+        history.replace('/settings/api-tokens?sort=name:ASC');
+        setCurrentStep('apiTokens.success');
```

Comment:
Can you please add tests? 🙏🏼

Reply:
I tried to use 
`await waitFor(() => {
      expect(history.location.pathname).toBe('/settings/transfer-tokens');
    });`
but unfortunately I am not able to trigger the redirect after the submission